modules/text_server_adv/text_server_adv.cpp

```diff
@@ -6114,6 +6117,11 @@ bool TextServerAdvanced::_shaped_text_update_breaks(const RID &p_shaped) {
 							continue;
 						}
 					}
+					// Do not add extra space for color picker object.
+					if (sd_glyphs[i].start == sd_glyphs[i].end || (i + 1 < sd->glyphs.size() && sd_glyphs[i + 1].start == sd_glyphs[i + 1].end)) {
```

Comment:
```suggestion
					if (((sd_glyphs[i].flags & GRAPHEME_IS_EMBEDDED_OBJECT) == GRAPHEME_IS_EMBEDDED_OBJECT && sd_glyphs[i].start == sd_glyphs[i].end) || (i + 1 < sd->glyphs.size() && (sd_glyphs[i + 1].flags & GRAPHEME_IS_EMBEDDED_OBJECT) == GRAPHEME_IS_EMBEDDED_OBJECT && sd_glyphs[i + 1].start == sd_glyphs[i + 1].end)) {
```

This condition is too generic and should not apply to anything except embedded objects.

Reply:
Ok. The same check is in like 1 or 2 other places. I'll update them there too